### ISSUE TITLE
Not working: Transfer of settings from 3.X version for apps set to not route though adguard

### ISSUE THREAD

**Issue by rodenti** [AdguardTeam/AdguardForAndroid#4657]
Upgrade from v3 to v4 beta 1 (4.0.788),   settings were not transferred for apps that were set to not route though adguard. 

To troubleshoot I restored the settings backup from v3 into v4,  and the apps not routing through adguard still aren't set up. I had to manually set up each app separately.

**Comment by jordanswrld94** [comment]
I think its because the whole app layout is different

**Comment by rodenti** [comment]
In don't think it is because of the app layout, although it's much tougher to find that app setting now. I usually go into Assistant from the notifications pull-down, pick the app from the recently opened list and choose app management. This is where I find the "Route traffic through Adguard" setting for that app, which I expected to be migrated from the previous version.

**Comment by rodenti** [comment]
I ended up uninstalling the beta and rolling back to the stable version.

Developers, hopefully you get this issue resolved for the next time I try v4.

**Comment by Versty** [comment]
@rodenti Thank you for the feedback. We have tried to reproduce the issue on Android 13, but the settings transfer for the apps seems to work fine on our side. Therefore we would like to know some addition information. 

First of all, are you talking about `Ad blocking`, `HTTPS filtering` and `Firewall` settings for the app? 
If so, please provide the details: 
- android version and device model 
- steps to reproduce the issue
- actual result (specify the problematic settings) 

Would be great if you could also provide your settings file from version 3.X. 
Please send the file to `devteam@adguard.com` and mention the `4657` number in the subject.

**Comment by rodenti** [comment]
Thanks for the reply.

I'm using the following to exclude apps from going through Adguard:

Adguard 3.6.11 hamburger menu, Apps Management, select an app (ie PayPal),
change the slider for AdGuard Protection to off. (See attached image)

Those settings didn't carry forward in the beta  with the initial
installation nor with importing of settings.

I have an old phone with Adguard installed and basically the same settings
as my current phone, so I'll try the beta version there. If I get the same
result I'll send the settings file from 3.X as well as the settings file
from the beta version.

**Comment by Versty** [comment]
@rodenti Any news? Have you managed to reproduce this issue on the second device?

**Comment by rodenti** [comment]
Thanks for following up. I tried it on my secondary device (Samsung S10+)
and as you mentioned it worked perfectly.

I haven't tried again on my primary device.

Roger

**Comment by rodenti** [comment]
I tried to install the non-beta v4.0 on my primary device again and got the
exact same issue as before. The apps set to not route through Adguard are
now routing through Adguard, which means I'm not confident that all of my
other settings have been carried over.

I had to roll back to v3 again to restore my settings.

Roger

**Comment by rodenti** [comment]
I decided to move forward with v4 anyway.

I ended up exporting the JSON file from both my old phone and my current
phone, merged some of the differences from the current phone into the old
phone file (mainly the license key and the custom filters), then imported
the merged JSON file into AdGuard version 4 on my current phone. It seems
to have picked up the settings correctly this time.

Something about the export from v3 on my current phone was not importing
correctly into v4.

Roger
